### PR TITLE
Run dtest upgrade tests

### PR DESCRIPTION
**Pull request by nastra**

### CODE REVIEW

File: build-scripts/cassandra-dtest-pytest.sh [hunk 1]
```diff
@@ -73,6 +74,8 @@ elif [ "${DTEST_TARGET}" = "dtest-offheap" ]; then
     pytest -vv --log-level="INFO" --use-vnodes --num-tokens=32 --use-off-heap-memtables --junit-xml=nosetests.xml --junit-prefix=${DTEST_TARGET} -s --cassandra-dir=$CASSANDRA_DIR --skip-resource-intensive-tests 2>&1 | tee -a ${WORKSPACE}/test_stdout.txt
 elif [ "${DTEST_TARGET}" = "dtest-large" ]; then
     pytest -vv --log-level="INFO" --use-vnodes --num-tokens=32 --junit-xml=nosetests.xml --junit-prefix=${DTEST_TARGET} -s --cassandra-dir=$CASSANDRA_DIR --only-resource-intensive-tests 2>&1 | tee -a ${WORKSPACE}/test_stdout.txt
+elif [ "${DTEST_TARGET}" = "dtest-upgrade" ]; then
+    pytest -vv --log-level="INFO" --execute-upgrade-tests --junit-xml=nosetests.xml --junit-prefix=${DTEST_TARGET} -s --cassandra-dir=$CASSANDRA_DIR 2>&1 | tee -a ${WORKSPACE}/test_stdout.txt
```
Comment (snazy) [inline build-scripts/cassandra-dtest-pytest.sh]: Does this need `--use-vnodes` ?

Comment (nastra) [reply build-scripts/cassandra-dtest-pytest.sh]: don't think so. The upgrade tests in CircleCI also test with vnodes disabled

**Comment by michaelsembwever** [comment]
Testing happening at https://ci-cassandra.apache.org/view/patches/job/Cassandra-devbranch-dtest-upgrade/3/console

**Review by snazy — APPROVED** [review]
ASF CI's still running after 12h though

File: build-scripts/cassandra-dtest-pytest.sh [hunk 2]
```diff
@@ -21,6 +21,7 @@ export CASSANDRA_DIR=${WORKSPACE}
 #Have Cassandra skip all fsyncs to improve test performance and reliability
 export CASSANDRA_SKIP_SYNC=true
 export TMPDIR="./tmp"
+export RUN_STATIC_UPGRADE_MATRIX=true
```
Comment (michaelsembwever) [inline build-scripts/cassandra-dtest-pytest.sh]: wonder if it is better/safer to move this line to inside just the case for "dtest-upgrade" (i.e. lines 76-77) ?

Comment (michaelsembwever) [reply build-scripts/cassandra-dtest-pytest.sh]: looks like the move is necessary, after running https://ci-cassandra.apache.org/view/patches/job/Cassandra-devbranch-dtest/146/console

Comment (nastra) [reply build-scripts/cassandra-dtest-pytest.sh]: done

**Review by michaelsembwever — APPROVED** [review]

**Comment by michaelsembwever** [comment]
merged with https://github.com/apache/cassandra-builds/commit/94f2b784fac3142164fc7fb667f132450a3fbfaa

**Pull request closed without merging**